docs/features/pouch_with_upgrade.md

```diff
@@ -0,0 +1,110 @@
+# Upgrade Container
+
+## Introduction
+
+In the cloud native world, developers were encouraged to build stateless apps, because stateful apps didn't scale well in the cloud's distributed. But there are still many  stateful apps, like database, desktop and operating systems appplication. Even in kubernetes, there is a [StatefulSet](https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/) component to manage the stateful applications.
+
+The stateful service upgrade requires that the new container must inherit all resources of old container, such as network and storage. The following two cases are the service provision scenarios of stateful containers:
+
+* Customer Case 1: In a database service, remote data is downloaded to the local computer as the initial data in the database when the container service is created for the first time. Database initialization lasts for a long time, so the new container must inherit the data stored on the old container in the consequent service upgrades to shorten the service provision time.
+
+* Customer Case 2: In a middleware service, the service registration mode is used. That is, the IP addresses of the scaled-up containers must be registered in the server list; otherwise, the containers are invalid. The new containers must inherit the IP addresses of old containers every time the service containers are upgraded; otherwise, the new services are invalid.
+
+How to make the upgrade procedure of stateful containers more simplified and more efficient that is what `upgrade` API of pouchd aims to achieve.
+
+## Design
+
+The following architecture graph shows how to upgrade a container:
+
+![pouch_with_upgrade](../static_files/pouch_with_upgrade.png)
```

Comment:
why not take three steps to explain this graph? 😃 

first, inherit container attributes from the old container in pouchd's memory;
second, replace container image via ......
third, just start the newly constructed command if needed.

Otherwise, people may not have sense of it. WDYT? @HusterWan